ArchiSteamFarm/Steam/Integration/ArchiWebHandler.cs

```diff
@@ -369,6 +369,27 @@ public sealed class ArchiWebHandler : IDisposable {
 			return result;
 		}
 
+		[PublicAPI]
+		public async Task<long?> GetPointBalance() {
+			Uri request = new(SteamStoreURL, "/pointssummary?l=english");
+
+			HtmlDocumentResponse? response = await UrlGetToHtmlDocumentWithSession(request, checkSessionPreemptively: false).ConfigureAwait(false);
+
+			if (response == null) {
+				return null;
+			}
+
+			IEnumerable<IElement>? htmlNodes = response.Content.SelectNodes("//*[contains(@class,'total_current')][last()]");
+
+			if (htmlNodes == null){
+				return null;
+			}
+
+			long myPoint = Convert.ToInt64(htmlNodes.First().TextContent.Replace(",", "", StringComparison.Ordinal), CultureInfo.CurrentCulture);
```

Comment:
There is a log going on which is wrong here.

1. The fact that you use `SelectNodes()` above just to do `.First()` here is awful. Just use `SelectSingleElementNode()` instead.
2. There is no validation of any kind whether the text is not null or empty, check other ASF occurances that refer to `TextContent` and try to follow them.
3. `Convert` is one of the worst possible ways to approach int parsing. You should use `long.TryParse()` instead, with all required checks for failure, once again check other ASF places for examples.
4. `CurrentCulture` is wrong here because you're parsing culture-agnostic string, it should be `InvariantCulture` instead.
5. I'm not sure about this, but it should be possible to use aforementioned `TryParse()` function with `NumberStyles.Currency` in order to parse the `,` without removing it firstly. Give it a try, if not then removing `,` as you did is alright.
6. Naming the variable `result` makes more sense as it's the result of the function.

Reply:
7. Is it possible for points balanace to be negative? Because if not, it should be `uint` instead. I also don't believe you need more than `2147483647` points, so `int` instead of `long` would suffice.

Reply:
I recommend 

> And here comes the real bomb of doing it properly:
> 
> You should call `https://store.steampowered.com/pointssummary/ajaxgetasyncconfig` to get the web api token:
> 
> ```json
> {"success":1,"data":{"webapi_token":"redacted"}}
> ```
> 
> Then use it in `https://api.steampowered.com/ILoyaltyRewardsService/GetSummary/v1/?access_token=<token>&steamid=<steamid>` request:
> 
> ```json
> {"response":{"summary":{"points":"33225","points_earned":"163625","points_spent":"130400"},"timestamp_updated":1621318615,"auditid_highwater":"1215917101"}}
> ```
> 
> And then you don't need to go through this awful HTML xPath whatsoever.

Thanks a lot, I tried use GetSummary API but it says the `key` is wrong, I get it now.

Reply:
@chr233 you're welcome, take your time. There are a lot of examples all over ASF in regards to how to deal with json data, so you shouldn't have too much of an issue.

Reply:
About the "negative point", I remember when the point-shop-system was just launched, the point balance can be negative. Rright now the steam will auto retake the items if someone's point turns to negative, i am not quite sure the value is always positive

Reply:
We can always assume it to be `uint` until somebody proves us wrong. If you say that it's not possible anymore, chances are nobody has negative anymore. If somebody comes up saying he does, we can easily change it to `int`.